pkg/kubelet/cri/remote/fake/endpoint.go

```diff
@@ -18,11 +18,21 @@ limitations under the License.
 
 package fake
 
+import (
+	"fmt"
+	"math/rand"
+	"time"
+)
+
 const (
-	defaultUnixEndpoint = "unix:///tmp/kubelet_remote.sock"
+	defaultUnixEndpoint = "unix:///tmp/kubelet_remote_%v.sock"
 )
 
 // GenerateEndpoint generates a new unix socket server of grpc server.
 func GenerateEndpoint() (string, error) {
-	return defaultUnixEndpoint, nil
+	// random sleep
+	time.Sleep(time.Duration(rand.Float64() * 10))
+
+	// use timestamp be a part fo file name
+	return fmt.Sprintf(defaultUnixEndpoint, time.Now().UnixNano()), nil
```

Comment:
could we replace the sleep with a random int?

Reply:
yeah. Using rand in `k8s.io/apimachinery/pkg/util/rand` also ok,  see updated.